### PR TITLE
fix: Add CHE_DEVFILE_REGISTRY_URL into devfile-registry configmap

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?
Adds `CHE_DEVFILE_REGISTRY_URL` environment variable into che-devfile-registry configmap

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20100

### How to test this PR?
- Change value of [eclipse-che-server dependency](https://github.com/che-incubator/chectl/blob/main/package.json#L28) to git://github.com/svor/che-server#sv/dr-url 
- Build chectl 
- Deploy Che via `run server:deploy --installer=helm --platform=minikube --multiuser`
- Check if icons are displayed and it's possible to create a workspace with sample

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: deploy/kubernetes/helm/che/custom-charts/che-devfile-registry/templates/configmap.yaml [hunk 1]
```diff
@@ -23,3 +23,4 @@ data:
   CHE_DEVFILE_IMAGES_REGISTRY_TAG: {{ . | quote }}
   {{- end }}
 {{- end }}
+  CHE_DEVFILE_REGISTRY_URL: {{ template "devfileRegistryUrl" . }}
```
Comment (tolusha) [inline deploy/kubernetes/helm/che/custom-charts/che-devfile-registry/templates/configmap.yaml:26]: We need something like this here https://github.com/eclipse-che/che-server/blob/main/deploy/kubernetes/helm/che/templates/configmap.yaml#L122-L132
```
{{- if .Values.che.workspace.devfileRegistryUrl }}
  CHE_WORKSPACE_DEVFILE__REGISTRY__URL: {{ .Values.che.workspace.devfileRegistryUrl | quote }}
{{- else }}
  CHE_WORKSPACE_DEVFILE__REGISTRY__URL: {{ template "devfileRegistryUrl" . }}
{{- end }}
```

Comment (svor) [reply deploy/kubernetes/helm/che/custom-charts/che-devfile-registry/templates/configmap.yaml:29]: fixed

**Comment by svor** [comment]
@tolusha could you please test these changes before merging, please follow the instructions from the description

File: deploy/kubernetes/helm/che/custom-charts/che-devfile-registry/templates/configmap.yaml (hunk 1 above)
Comment (tolusha) [reply deploy/kubernetes/helm/che/custom-charts/che-devfile-registry/templates/configmap.yaml:29]: to check if there is no issues with templates, pls use

```
cd <CHE_SERVER_REPOSITORY>/deploy/kubernetes/helm/che
helm dependency update
chectl server:deploy --platform minikube --installer helm -m --templates <CHE_SERVER_REPOSITORY>/deploy/
```

**Comment by svor** [comment]
Have a problem with getting `Values.che.workspace.devfileRegistryUrl`
```
Error: template: che/charts/che-devfile-registry/templates/deployment.yaml:35:28: executing "che/charts/che-devfile-registry/templates/deployment.yaml" at <include (print $.Template.BasePath "/configmap.yaml") .>: error calling include: template: che/charts/che-devfile-registry/templates/configmap.yaml:26:14: executing "che/charts/che-devfile-registry/templates/configmap.yaml" at <.Values.che.workspace.devfileRegistryUrl>: nil pointer evaluating interface {}.workspace
```

**Comment by tolusha** [comment]
Probably because that templates is located in a parent directory.

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by svor** [comment]
close it in favor of https://github.com/eclipse-che/che-devfile-registry/pull/441

**Pull request closed without merging**